Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/BPMNEditorController.java

```diff
@@ -231,6 +232,39 @@ public void onEvent(final Event event) throws InterruptedException {
       }
     });
 
+    this.addEventListener("onSimulateModel", new EventListener<Event>() {
+      @Override
+      public void onEvent(final Event event) throws InterruptedException {
+        PortalContext portalContext = mainC.getPortalContext();
+        Map<String, PortalPlugin> portalPluginMap = portalContext.getPortalPluginMap();
+        PortalPlugin simulateModelPlugin = portalPluginMap.get(PluginCatalog.PLUGIN_SIMULATE_MODEL);
+
+        //Since simulate model is an EE feature, it may not be available
+        if (simulateModelPlugin == null) {
+          Messagebox.show(Labels.getLabel("portal_simModelUnavailable_message"),
+                  Labels.getLabel("portal_simModelUnavailable_title"),
+                  Messagebox.OK, Messagebox.INFORMATION);
+          return;
+        }
+
+        if (currentUserAccessType == AccessType.VIEWER) {
+          Notification.error(Labels.getLabel("portal_noPrivilegeSaveEdit_message"));
+          return;
+        }
+
+        //Only allow model to be simulated if it is saved
+        if (isNewProcess || process == null) {
+          Notification.error(Labels.getLabel("portal_saveModelFirst_message"));
+        } else {
```

Comment:
Checking for new process here is not necessary because the simulation can work on unsaved model, we can remove this check without problems. Note that it's different from "onShare" event handling because it makes no sense to share an unsaved model (a share record will be created in the database with reference to the saved model, thus a new model must  be saved first to have a record in the database).

Reply:
Removed check for new/saved process. Logs created by simulating unsaved models now have a suggested name of "untitled_sim_scenario".